PGM/src/main/java/tc/oc/pgm/map/MapPersistentContext.java

```diff
@@ -0,0 +1,48 @@
+package tc.oc.pgm.map;
+
+import com.google.common.collect.Range;
+import org.jdom2.Document;
+import tc.oc.api.docs.SemanticVersion;
+import tc.oc.pgm.modules.InfoModule;
+
+/**
+ * This is a replacement for classes that need information from MapModuleContext to use. MapModuleContext will not be
+ * guaranteed to be available if a match isn't being played for the map, but MapPersistentContext will always contain
+ * basic information required to reference the map (like map name, authors, or player amount).
+ */
+public class MapPersistentContext {
+
+    private SemanticVersion proto;
+    private Range<Integer> playerLimits;
+    private InfoModule infoModule;
+    private MapDocument apiDocument;
+
+    public MapPersistentContext(MapModuleContext context) {
+        this.proto = context.getProto();
+        this.playerLimits = context.playerLimits();
+        this.infoModule = context.needModule(InfoModule.class);
```

Comment:
Can we directly inject the `InfoModule`? The `needModule` method is actually a getter from `ModuleContext`, which appears to be an old system.

Reply:
That would potentially complicate things as this object is created just after reloading a map, from the module context, if we were to create this via injection, MapDefinition would need providers for both MapModuleContext and MapPersistentContext.
Still then i wouldn't use the context as param and would just inject the proto and info module as params, and the MapModuleContext injected as part of the constructor to extract player limits and apiDoc

Reply:
But, it does make sense for the MapDefinition to provision MapPersistentContext, as well as the MapModuleContext. I could see it being useful in making it injection-friendly and stepping away from relics. It does look relatively loose from the rest of the classes in the package and doesn't really serve any purpose other than providing the InfoModule. In fact, is it possible if we store the InfoModule somewhere, say the MapDefinition so we are also guaranteed that the InfoModule is present as long as the MapDefinition is loaded?

Reply:
`But, it does make sense for the MapDefinition to provision MapPersistentContext, as well as the MapModuleContext` No, that's not that MapDefinition privovisions those classes, is that it needs Providers to create those classes when required aka: a `Provider<MapModuleContext>` is injected into MapDefinition, and on reload, the provider is called to get an instance of MapModuleContext.

If we store the InfoModule in the map definition we are linking the two together way too much. The MapDefinition shouldn't directly hold a reference to a specific PGM module.